src/framework/audio/main/audiomodule.cpp

```diff
@@ -317,37 +320,41 @@ void AudioModule::setupAudioWorker(const IAudioDriver::Spec& activeSpec)
     consts.minSamplesToReserveWhenIdle = m_configuration->minSamplesToReserve(RenderMode::IdleMode);
     consts.minSamplesToReserveInRealtime = m_configuration->minSamplesToReserve(RenderMode::RealTimeMode);
 
-    auto workerSetup = [this, activeSpec, consts]() {
+    std::shared_ptr<worker::AudioBuffer> audioBuffer = m_workerModule->audioBuffer();
+
+    auto workerSetup = [this, activeSpec, consts, audioBuffer]() {
         AudioSanitizer::setupWorkerThread();
         ONLY_AUDIO_WORKER_THREAD;
 
+        //! NOTE It should be as early as possible
+        m_rpcChannel->initOnWorker();
+
         m_workerModule->onPreInit(IApplication::RunMode::GuiApp);
 
         // Setup audio engine
         std::shared_ptr<worker::AudioEngine> audioEngine = m_workerModule->audioEngine();
-        audioEngine->init(m_audioBuffer, consts);
+        audioEngine->init(audioBuffer, consts);
         audioEngine->setAudioChannelsCount(m_configuration->audioChannelsCount());
         audioEngine->setSampleRate(activeSpec.sampleRate);
         audioEngine->setReadBufferSize(activeSpec.samples);
 
         audioEngine->setOnReadBufferChanged([this](const samples_t samples, const sample_rate_t rate) {
             msecs_t interval = m_configuration->audioWorkerInterval(samples, rate);
-            m_audioWorker->setInterval(interval);
+            m_audioThread->setInterval(interval);
         });
 
         m_workerModule->synthResolver()->init(m_configuration->defaultAudioInputParams());
 
-        m_rpcChannel->initOnWorker();
-
         m_workerModule->onInit(IApplication::RunMode::GuiApp);
     };
 
-    auto workerLoopBody = [this]() {
+    auto workerLoopBody = [this, audioBuffer]() {
```

Comment:
I'm not sure that capturing the buffer is a good idea... Because at first glance, it seems that the worker module owns the pointer, but in practice, it is still the audio module. Maybe we should keep it as a member of the audio module?